### ISSUE TITLE
Segmentation Fault When Called from Command Line on Mac

### ISSUE THREAD

**Issue by qpding** [darktable-org/darktable#17671]
### Describe the bug

When calling darktable with the following command:
`/Applications/darktable.app/Contents/MacOS/darktable -d lua --library :memory: --conf plugins/lighttable/export/icctype=3 --conf "lua/export_on_exit/export_filename=/Users/quinn/Downloads/export.jpeg" "test_raw.ARW"`
Then double click the image to enter darkroom for editing, darktable gives a segmentation fault and the program is terminated.
However, sometimes (very randomly and rarely), it does enter the darkroom and I was able to conduct editing. When that happens, there is still a buffer alignment problem warning.

### Steps to reproduce

1. Launch terminal (in my case iTerm2 and zsh)
2. Run this command: /Applications/darktable.app/Contents/MacOS/darktable -d lua --library :memory: --conf plugins/lighttable/export/icctype=3 --conf "lua/export_on_exit/export_filename=/Users/quinn/Downloads/export.jpeg" "test_raw.ARW"
3. Double click the image in darktable UI
4. It either quits giving segmentation fault error, or enters the darkroom but still gives a buffer alignment problem warning.

### Expected behavior

darktable enters the darkroom for editing without error

### Logfile | Screenshot | Screencast

When darktable gets terminated, the full log is:
```
/Applications/darktable.app/Contents/MacOS/darktable -d lua --library :memory: --conf plugins/lighttable/export/icctype=3 --conf "lua/export_on_exit/export_filename=/Users/quinn/Downloads/export.jpeg" "test_raw.ARW"
darktable 4.8.1
Copyright (C) 2012-2024 Johannes Hanika and other contributors.

Compile options:
  Bit depth              -> 64 bit
  Debug                  -> DISABLED
  SSE2 optimizations     -> DISABLED
  OpenMP                 -> ENABLED
  OpenCL                 -> ENABLED
  Lua                    -> ENABLED  - API version 9.3.0
  Colord                 -> DISABLED
  gPhoto2                -> ENABLED
  GMIC                   -> ENABLED  - Compressed LUTs are supported
  GraphicsMagick         -> DISABLED
  ImageMagick            -> ENABLED
  libavif                -> ENABLED
  libheif                -> ENABLED
  libjxl                 -> ENABLED
  OpenJPEG               -> ENABLED
  OpenEXR                -> ENABLED
  WebP                   -> ENABLED

See https://www.darktable.org/resources/ for detailed documentation.
See https://github.com/darktable-org/darktable/issues/new/choose to report bugs.


(process:20407): GLib-GObject-CRITICAL **: 18:57:04.100: g_object_set: assertion 'G_IS_OBJECT (object)' failed

(darktable:20407): Gtk-CRITICAL **: 18:57:04.339: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
2024-10-15 18:57:07.124 darktable[20407:525917] +[IMKClient subclass]: chose IMKClient_Legacy
2024-10-15 18:57:07.124 darktable[20407:525917] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
     3.2083 LUA ERROR : cannot open /Users/quinn/.config/darktable/luarc: No such file or directory
[1]    20407 segmentation fault  /Applications/darktable.app/Contents/MacOS/darktable -d lua --library :memory
```
When sometimes it does not get terminated, the log is:
```
(process:20420): GLib-GObject-CRITICAL **: 18:57:23.746: g_object_set: assertion 'G_IS_OBJECT (object)' failed

(darktable:20420): Gtk-CRITICAL **: 18:57:23.935: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
2024-10-15 18:57:26.544 darktable[20420:526491] +[IMKClient subclass]: chose IMKClient_Legacy
2024-10-15 18:57:26.544 darktable[20420:526491] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
     2.9651 LUA ERROR : cannot open /Users/quinn/.config/darktable/luarc: No such file or directory
     8.8387 [pixelpipe_process CPU] buffer aligment problem: IN=0x174b40040 OUT=0x309d08000
     9.3490 [pixelpipe_process CPU] buffer aligment problem: IN=0x130858040 OUT=0x12afc8000
```

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.8.1

### What OS are you using?

Mac

### What is the version of your OS?

MacOS 15.0.1

### Describe your system?

1. darktable is installed with homebrew using their latest formula on Oct. 14 2024.
2. My computer is MacBook Air with M2 chip and 16G RAM.

### Are you using OpenCL GPU in darktable?

I dont know

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Here is a test raw image from Sony alpha 6400. But it may not really matter because it also happens with raw files from Sony A7iv.
[test_raw.zip](https://github.com/user-attachments/files/17377192/test_raw.zip)

**Comment by wpferguson** [comment]
Was it working before you used the latest homebrew recipe?

**Comment by zisoft** [comment]
> With both 4.8.0 and 4.6.1, the darktable UI goes directly to the darkroom view when called from command line. However with 4.8.1, it goes to the lighttable view.

This is a restriction we have to live with. Command line processing has changed in master and on macOS it is not possible to distinguish whether a single file or multiple files are passed via command line, so we stay in lighttable now.

**Comment by zisoft** [comment]
@qpding: I cannot reproduce on current master and the crash on import via command line on macOS was fixed with #17298.

Can you test with a [nightly build](https://github.com/darktable-org/darktable/releases)?

**Comment by qpding** [comment]
> @qpding: I cannot reproduce on current master and the crash on import via command line on macOS was fixed with #17298.
> 
> Can you test with a [nightly build](https://github.com/darktable-org/darktable/releases)?

Hmm... Interesting, it works well with nightly build 20241020 (darktable-4.9.0+892.g40f497b425-arm64.dmg), the crash is no longer reproducible. Thanks for the help!

Don't know if I should but please close this issue if it is already a solved problem.

Here's the log if it can help with anything. And the _**buffer aligment problem**_ is no longer present.
```
/Applications/darktable.app/Contents/MacOS/darktable -d lua --library :memory: --conf plugins/lighttable/export/icctype=3 --conf "lua/export_on_exit/export_filename=/Users/quinn/Downloads/export.jpeg" "test_raw.ARW"
darktable 4.9.0+892~g40f497b425
Copyright (C) 2012-2024 Johannes Hanika and other contributors.

Compile options:
  Bit depth              -> 64 bit
  Debug                  -> DISABLED
  SSE2 optimizations     -> DISABLED
  OpenMP                 -> ENABLED
  OpenCL                 -> ENABLED
  Lua                    -> ENABLED  - API version 9.4.0
  Colord                 -> DISABLED
  gPhoto2                -> ENABLED
  GMIC                   -> ENABLED  - Compressed LUTs are supported
  GraphicsMagick         -> DISABLED
  ImageMagick            -> ENABLED
  libavif                -> ENABLED
  libheif                -> ENABLED
  libjxl                 -> ENABLED
  LibRaw                 -> ENABLED  - Version 0.22.0-Devel202403
  OpenJPEG               -> ENABLED
  OpenEXR                -> ENABLED
  WebP                   -> ENABLED

See https://www.darktable.org/resources/ for detailed documentation.
See https://github.com/darktable-org/darktable/issues/new/choose to report bugs.

     0.0001 [dt starting] /Applications/darktable.app/Contents/MacOS/darktable -d lua --library :memory: --conf plugins/lighttable/export/icctype=3 --conf lua/export_on_exit/export_filename=/Users/quinn/Downloads/export.jpeg test_raw.ARW

(process:68674): GLib-GObject-CRITICAL **: 14:37:29.609: g_object_set: assertion 'G_IS_OBJECT (object)' failed

(darktable:68674): Gtk-CRITICAL **: 14:37:29.992: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed
2024-10-20 14:37:31.968 darktable[68674:1907279] +[IMKClient subclass]: chose IMKClient_Legacy
2024-10-20 14:37:31.968 darktable[68674:1907279] +[IMKInputSession subclass]: chose IMKInputSession_Legacy
     2.4355 LUA ERROR : cannot open /Users/quinn/.config/darktable/luarc: No such file or directory
```
